packages/headless/src/features/insight-search/insight-search-actions-thunk-processor.test.ts

```diff
@@ -164,4 +164,107 @@ describe('AsyncInsightSearchThunkProcessor', () => {
     });
     expect(processed.automaticallyCorrected).toBe(true);
   });
+
+  describe('query correction processing', () => {
+    describe('legacy query correction processing', () => {
+      it('should correct the query the legacy way', async () => {
```

Comment:
that's not clear enough, what's the legacy,
The goal is to understand the expected behaviour just by reading the test description.